client/state/wp-api/middleware.js

```diff
@@ -0,0 +1,22 @@
+import get from 'lodash/get';
+import isFunction from 'lodash/isFunction';
+
+import {
+	REQUIRE_POST
+} from 'state/action-types';
+
+import { fetchPost } from './posts/fetch-post';
+
+const actionMap = {
+	[ REQUIRE_POST ]: fetchPost
```

Comment:
I was thinking something like `CREATE_POST`, `UPDATE_POST`, `DELETE_POST` (or with the words flipped as per our style).

The only real hiccup here is clashing with existing action types. I hesitate to suggest prepending `API_` on them because that's what we're getting away from.